lib/tasks/licenses.rake

```diff
@@ -14,19 +16,18 @@ namespace :licenses do  # rubocop:disable Metrics/BlockLength
 
   desc 'Check license compliance of dependencies'
   task :compliance do
-    warn 'Checking license action items...'
-    unless cli.action_items
-      warn '*** License compliance test failed  ***'
-      exit 1
-    end
+    require 'license_finder'
+
+    warn 'Checking action items...'
+    exit(1) unless LicenseFinder::CLI::Main.new.action_items
```

Comment:
what is this doing?

Reply:
Action items is supposed to print a list of licenses that are to be approved. If we ever add a library with a weird license I guess this will warn us about it.

Reply:
But the shouldn't it be something like `exit(1) if LicenseFinder::CLI::Main.new.action_items.present?`.

Also Perhaps print action items out but this can be for another time.

Reply:
Unless I've got it wrong, which is highly possible, `#action_items` will print everything and return an IO object. Otherwise it will return nothing. I may be wrong, I am not sure how to reproduce it.

Reply:
To make it fail you just need to remove any approved license (e.g. MIT) from `dependency_decisions.yml`. I removed MIT, ran `rake licenses:compliance` and saw this error:

```
Dependencies that need approval:
3scale_client, 2.11.0, MIT
@babel/code-frame, 7.18.6, MIT
@babel/compat-data, 7.14.7, MIT
@babel/compat-data, 7.19.0, MIT
@babel/core, 7.14.6, MIT
@babel/generator, 7.19.0, MIT
...
```

`LicenseFinder::CLI::Main.new.action_items` exits the task when it finds a not-approved license, it returns nothing otherwise:

https://github.com/pivotal/LicenseFinder/blob/master/lib/license_finder/cli/main.rb#L133-L147

So, between `if ... .present?` and `unless`, the right answer is: it doesn't matter, you can just call it directly
```suggestion
    LicenseFinder::CLI::Main.new.action_items
```
It will `exit(1)` when not complying and continue to the up-to-date test otherwise.

Reply:
Wait... So `LicenseFinder::CLI::Main.new.action_items` will `exit(0)` if there are no dependencies to be approved and `exit(1)` if there's any. I don't think checking the value it returns will work. 
It must be returning an object of `<IO:<STDOUT>>`, which should always be present 🤔 Like `STDOUT.present? #=> true`.
Also, as this `action_items` always exits, I'd expect this rake task to exit as well when `action_items` is called...
Maybe we should call this method with a rescue for `SystemExit`. Then we can check the exit code status with something like like:
```ruby
LicenseFinder::CLI::Main.new.action_items
rescue SystemExit => error
  error.status #=> should be 0 or 1
end
```

Reply:
After further investigation I've found out that `LicenseFinder::CLI::Main.new.action_items` will abort the rake task if there are any action items. IMHO it doesn't make sense to check the returned `IO:STDOUT`, because if the task doesn't stop it means there aren't any pending actions items 😄 .